scripts/make.go

```diff
@@ -236,11 +236,17 @@ func buildFlags() []string {
 	if err != nil {
 		log.Fatal(err)
 	}
+	gctrimpath := "-gcflags=all=-trimpath=${PWD}"
+	asmtrimpath := "-asmflags=all=-trimpath=${PWD}"
 	ldFlags := "-X main.Build=" + strings.TrimSpace(string(buildSHA))
 	if runtime.GOOS == "darwin" {
 		ldFlags = "-s " + ldFlags
 	}
-	return []string{fmt.Sprintf("-ldflags=%s", ldFlags)}
+	extldFlags := os.Getenv("LDFLAGS")
+	if extldFlags != "" {
+		ldFlags = ldFlags + fmt.Sprintf(" -extldflags %s", extldFlags)
```

Comment:
I think delve is usually linked internally and this won't do anything (usually).

Reply:
You'd think so, but we have been trying to compile all go packages on Arch with full RELRO, and funny enough this is needed for this.

```
λ delve foxboron/fix-script» go build -i -v -buildmode=pie -o dlv ./cmd/dlv 
λ delve foxboron/fix-script» checksec -f dlv                               
RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH	Symbols		FORTIFY	Fortified	Fortifiable  FILE
Partial RELRO   Canary found      NX enabled    PIE enabled     No RPATH   No RUNPATH   19406 Symbols	^C
λ delve foxboron/fix-script» go build -i -v -buildmode=pie -ldflags "-extldflags $LDFLAGS" -o dlv ./cmd/dlv 
λ delve foxboron/fix-script» checksec -f dlv                                                               
RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH	Symbols		FORTIFY	Fortified	Fortifiable  FILE
Full RELRO      Canary found      NX enabled    PIE enabled     No RPATH   No RUNPATH   19405 Symbols	
```

Reply:
Sure, if you pass buildmode=pie, but make.go isn't.

Reply:
Yes. Arch personally hacks this by patching `go-pie`. However, extra parsing is required  to utilize `GOFLAGS`, so it's easier for any distribution involved to just pass `LDFLAGS` directly.

Reply:
That's weird, but ok.

Reply:
`export GOFLAGS="-ldflags=-extldflags=-zrelro -ldflags=-extldflags=-znow"` Is essentially the format.

`export GOFLAGS="-ldflags=-extldflags=$LDFLAGS"` will result in the `GOFLAGS` parser breaking on you. So things get tedious as they are not following the CLI format.